lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -151,7 +151,11 @@ public function lastInsertId($name = null)
             return false;
         }
 
-        OraclePlatform::assertValidIdentifier($name);
+        $identifiers = explode('.', $name);
+
+        foreach ($identifiers as $identifier) {
```

Comment:
`"foo"."bar"."baz"."tab"` is most probably invalid. Probably already invalid with > 2 levels